Make gst-switch warning free with -Wall and enabled -Werror

```
tansell@tansell-x1c-l:~/foss/gst-switch/gst-switch-src$ make -j8
make  all-recursive
make[1]: Entering directory `/home/tansell/foss/gst-switch/gst-switch-src'
Making all in plugins
make[2]: Entering directory `/home/tansell/foss/gst-switch/gst-switch-src/plugins'
  CC       libgstswitch_la-gstswitchplugin.lo
  CC       libgstswitch_la-gsttcpmixsrc.lo
  CC       libgstswitch_la-gstswitch.lo
  CC       libgstswitch_la-gstconvbin.lo
  CC       libgstassess_la-gstassessplugin.lo
  CC       libgstassess_la-gstassess.lo
  CCLD     libgstassess.la
  CCLD     libgstswitch.la
make[2]: Leaving directory `/home/tansell/foss/gst-switch/gst-switch-src/plugins'
Making all in tools
make[2]: Entering directory `/home/tansell/foss/gst-switch/gst-switch-src/tools'
  CC       gst_switch_srv-gstworker.o
  CC       gst_switch_srv-gstswitchserver.o
  CC       gst_switch_srv-gstcase.o
  CC       gst_switch_srv-gstcomposite.o
  CC       gst_switch_srv-gstswitchcontroller.o
  CC       gst_switch_srv-gstrecorder.o
  CC       gst_switch_srv-gsocketinputstream.o
  CC       gst_switch_ui-gstworker.o
gstswitchserver.c: In function ‘gst_switch_server_parse_args’:
gstswitchserver.c:174:5: warning: format ‘%s’ expects argument of type ‘char *’, but argument 4 has type ‘char **’ [-Wformat=]
     ERROR ("unknown option: %s", argv[1]);
     ^
  CC       gst_switch_ui-gstswitchui.o
  CC       gst_switch_ui-gstvideodisp.o
  CC       gst_switch_ui-gstaudiovisual.o
  CC       gst_switch_ui-gstswitchclient.o
  CC       gst_switch_cap-gstworker.o
  CC       gst_switch_cap-gstswitchcapture.o
  CC       gst_switch_cap-gstswitchclient.o
gstaudiovisual.c: In function ‘gst_audio_visual_message’:
gstaudiovisual.c:348:7: warning: ‘g_value_array_get_type’ is deprecated (declared at /usr/include/glib-2.0/gobject/gvaluearray.h:60): Use 'GArray' instead [-Wdeprecated-declarations]
       } else if (G_VALUE_HOLDS (list_rms, G_TYPE_VALUE_ARRAY)) {
       ^
gstaudiovisual.c:353:11: warning: ‘g_value_array_get_nth’ is deprecated (declared at /usr/include/glib-2.0/gobject/gvaluearray.h:63): Use 'GArray' instead [-Wdeprecated-declarations]
           value = g_value_array_get_nth (va, i);
           ^
gstswitchui.c: In function ‘gst_switch_ui_init’:
gstswitchui.c:285:3: warning: ‘gtk_scrolled_window_add_with_viewport’ is deprecated (declared at /usr/include/gtk-3.0/gtk/gtkscrolledwindow.h:125): Use 'gtk_container_add' instead [-Wdeprecated-declarations]
   gtk_scrolled_window_add_with_viewport (GTK_SCROLLED_WINDOW (scrollwin),
   ^
  CCLD     gst-switch-srv
  CCLD     gst-switch-cap
  CCLD     gst-switch-ui
make[2]: Leaving directory `/home/tansell/foss/gst-switch/gst-switch-src/tools'
Making all in tests/unit
make[2]: Entering directory `/home/tansell/foss/gst-switch/gst-switch-src/tests/unit'
make  all-am
make[3]: Entering directory `/home/tansell/foss/gst-switch/gst-switch-src/tests/unit'
make[3]: Nothing to be done for `all-am'.
make[3]: Leaving directory `/home/tansell/foss/gst-switch/gst-switch-src/tests/unit'
make[2]: Leaving directory `/home/tansell/foss/gst-switch/gst-switch-src/tests/unit'
Making all in tests
make[2]: Entering directory `/home/tansell/foss/gst-switch/gst-switch-src/tests'
  CC       test_switch_server-test_switch_server.o
  CC       test_switch_server-gstworker.o
  CC       test_switch_server-gstswitchclient.o
  CC       test_fd_leaks-test_fd_leaks.o
test_switch_server.c: In function ‘child_quit’:
test_switch_server.c:331:41: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     children = g_list_remove (children, (gconstpointer) pid);
                                         ^
test_switch_server.c: In function ‘launch’:
test_switch_server.c:450:41: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
     children = g_list_append (children, (gpointer) pid);
                                         ^
test_switch_server.c: In function ‘kill_children’:
test_switch_server.c:2372:16: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
     GPid pid = (GPid) child->data;
                ^
  CCLD     test-fd-leaks
  CCLD     test-switch-server
make[2]: Leaving directory `/home/tansell/foss/gst-switch/gst-switch-src/tests'
make[2]: Entering directory `/home/tansell/foss/gst-switch/gst-switch-src'
make[2]: Leaving directory `/home/tansell/foss/gst-switch/gst-switch-src'
make[1]: Leaving directory `/home/tansell/foss/gst-switch/gst-switch-src'
```

--- a/configure.ac
+++ b/configure.ac
@@ -129,6 +129,17 @@ AC_COMPILE_IFELSE([AC_LANG_PROGRAM([ ], [ ])], [
   AC_MSG_RESULT([no])
 ])
 
+dnl check if compiler understands -Werror (if yes, add -Werror to GST_CFLAGS)
+AC_MSG_CHECKING([to see if compiler understands -Werror])
+save_CFLAGS="$CFLAGS"
+CFLAGS="$CFLAGS -Werror"
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([ ], [ ])], [
+  GST_CFLAGS="$GST_CFLAGS -Werror"
+  AC_MSG_RESULT([yes])
+], [
+  AC_MSG_RESULT([no])
+])
+
 dnl set the plugindir where plugins should be installed (for plugins/Makefile.am)
 if test "x${prefix}" = "x$HOME"; then
   plugindir="$HOME/.gstreamer-1.0/plugins"

